pkg/credentialprovider/aws/aws_credentials.go

```diff
@@ -161,6 +238,10 @@ func newEcrProvider(region string, getter tokenGetter) *ecrProvider {
 // TODO: figure how to enable it manually for deployments that are not on AWS but still
 // use ECR somehow?
 func (p *ecrProvider) Enabled() bool {
```

Comment:
@mcrute would it be possible to register `p.region` here if the region is not already in the list of well-known regions (according to SDK) without checking metadata for the current region? Can we assume that the region requested by the credential provider is valid? That would remove the need to query metadata at all. 

If checking metadata is needed, can we query it in `Enabled()` within a `once.Do`?

Reply:
Hmmm.. would that not work because you have to register the region first before the URL can be pulled?

Reply:
It would probably work but doesn't fit with the flow of the code. `NewDockerKeyring` loops over all registered credential providers and adds them to the live providers list if they're enabled. Registering new providers as a side-effect of checking if existing ones are enabled seems sub-optimal.

The way ECR and IAM work it would be possible to obtain only one ECR login token from the current region and use it to fetch images from any region. The problem with that is it's not backwards compatible because it requires people to trust a cross-region principal in their IAM policies rather than the one in the current region.

Another possible option would be to register a glob pattern `*.dkr.ecr.*.*` and then parse the URL at call-time to determine the correct region from which to fetch the login token. That would allow us to register a single credential provider and eliminate a lot of the work we're doing today. That's basically a complete re-write of the AWS provider though so I'd prefer to handle that elsewhere.